### DISCUSSION TITLE
MutationTrigger type

### DISCUSSION THREAD

**Discussion by vbelinschi** [reduxjs/redux-toolkit#2711] [Q&A] [answered]
Hello

In our project we have multiple mutations, that are handled more or less the same way. Until now, we had a separate method for each mutation, something like this: 
```typescript
const [updatePasswordCall] = useUpdatePasswordMutation();
const resetPassword = async (mailAccountId: string, newPassword: string) => {
 showInfoNotification();
 await updatePasswordCall({
   customerNumber: customerIds.customerNumber,
   accountNumber: customerIds.accountNumber,
   mailId: mailAccountId,
   data: { password: newPassword },
 })
 .unwrap()
 .then(() => {
  showSuccessNotification();
 })
 .catch((error) => {
  showErrorNotification(error);
 });
};
```
It is especially a problem when we use multiple mutations in the same file, we have to repeat the above code for each and every mutation. 
I was thinking to make method, where I could pass the mutation trigger as an argument, and then call `.unwrap()` on it, and handle the result.
But when I tried to find the type of the mutation trigger, I found it's a
```typescript
MutationTrigger<MutationDefinition<DataType, BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError, {}, FetchBaseQueryMeta>, never, void, 'reducer-patch'>>;
```
and that is ok I guess, but the types are being imported from `'@reduxjs/toolkit/dist/query';` and I'm not comfortable to import things from dist folder.

Any suggestions on how to handle multiple mutation triggers the same way?

Thanks

**Comment by phryneas** [comment, marked as answer]
My first thought would be
```js
function handle(triggerResult: { unwrap(): Promise<any> }) {
  return triggerResult
  .unwrap()
 .then(() => {
  showSuccessNotification();
 })
 .catch((error) => {
  showErrorNotification(error);
 });
}
```

to be used as
```js
 await handle(updatePasswordCall({
   customerNumber: customerIds.customerNumber,
   accountNumber: customerIds.accountNumber,
   mailId: mailAccountId,
   data: { password: newPassword },
 }))
```

Generally: you don't need the exact trigger type. Just write a function with the partial type that actually interests you (in this case I want the object to have a `.unwrap` function that returns a `Promise`.

**Reply by vbelinschi** [reply]
Perfect. Thanks a lot
